File: leetcode1797.java
package org.example;

import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * @author 20179
 */
public class Main {

    public static void main(String[] args) throws IOException {

    }
}

class AuthenticationManager {
    private int timeToLive;
    private Map<String, Integer> keyMap;

    private void timeCheck(int currentTime) {
        List<String> list = new ArrayList<>();
        for (Map.Entry<String, Integer> entry : keyMap.entrySet()) {
            String k = entry.getKey();
            Integer v = entry.getValue();
            if (v <= currentTime) {
                list.add(k);
            }
        }
        list.forEach(keyMap::remove);
    }

    public AuthenticationManager(int timeToLive) {
        this.timeToLive = timeToLive;
        this.keyMap = new HashMap<>();
    }

    public void generate(String tokenId, int currentTime) {
        keyMap.put(tokenId, currentTime + timeToLive);
    }

    public void renew(String tokenId, int currentTime) {
        timeCheck(currentTime);
        if (keyMap.containsKey(tokenId)) {
            keyMap.put(tokenId,currentTime + timeToLive);
        }
    }

    public int countUnexpiredTokens(int currentTime) {
        timeCheck(currentTime);
        return keyMap.size();
    }
}

/**
 * Your AuthenticationManager object will be instantiated and called as such:
 * AuthenticationManager obj = new AuthenticationManager(timeToLive);
 * obj.generate(tokenId,currentTime);
 * obj.renew(tokenId,currentTime);
 * int param_3 = obj.countUnexpiredTokens(currentTime);
 */

File: leetcode1797.cpp
#include "bits/stdc++.h"
using namespace std;
class AuthenticationManager {
public:
    int timeToLive;
    map<string,int> keyMap;
    AuthenticationManager(int timeToLive) {
        this->timeToLive = timeToLive;
    }
    void TimeCheck(int currentTime) {
        for (auto &[k,v] : keyMap) {
            if (v <= currentTime) {
                keyMap.
            }
        }
    }
    void generate(string tokenId, int currentTime) {
        keyMap[tokenId] = currentTime + timeToLive;
    }

    void renew(string tokenId, int currentTime) {
        TimeCheck(currentTime);

    }

    int countUnexpiredTokens(int currentTime) {

    }
};

/**
 * Your AuthenticationManager object will be instantiated and called as such:
 * AuthenticationManager* obj = new AuthenticationManager(timeToLive);
 * obj->generate(tokenId,currentTime);
 * obj->renew(tokenId,currentTime);
 * int param_3 = obj->countUnexpiredTokens(currentTime);
 */
int main() {
    return 0;
}